daemon/audit.c

```diff
@@ -347,12 +388,20 @@ audit_write_file(const uuid_t *uuid, const AuditRecord *msg)
 	}
 
 	TRACE("Logging audit record to file: %s", file);
+	if (!audit_log_check_delimiter(file)) {
+		TRACE("Fixup audit delimiter to %s", file);
+		if (0 > file_write_append(file, AUDIT_DELIMITER, strlen(AUDIT_DELIMITER))) {
```

Comment:
May we should ensure the Delimiter was correctly written by  strlen(AUDIT_DELIMITER) == file_write_append(file, AUDIT_DELIMITER, strlen(AUDIT_DELIMITER) ?